DeliveryApi/ApiHandlers/src/com/ilmlf/delivery/api/handlers/PopulateFarmDb.java

```diff
@@ -88,47 +88,76 @@
   }
 
   /**
-   * Entry point for Custom Resource call. This function executes dbinit.sql and
-   * creates a user with the username and password from DB_SECRET.
+   * Entry point for Custom Resource call. This function executes dbinit.sql, which 
+   * creates a db user with username and password from DB_SECRET.
+   * 
+   * The Sql script will only execute on CREATE/UPDATE requestTypes 
+   * See https://docs.aws.amazon.com/cdk/api/latest/java/software/amazon/awscdk/customresources/package-summary.html 
    *
    * @param event Event object from CloudFormation Custom Resource
    * @param context Context object
    * @return Result to Custom Resource
    */
-  @Override
-  public String handleRequest(Object event, Context context) {
-    String script = "./com/ilmlf/db/dbinit.sql";
-    try (FileReader fr = new FileReader(script);
-         BufferedReader br = new BufferedReader(fr);
-         Statement stmt = con.createStatement()){
-
-      StringBuilder sb = new StringBuilder();
-      String line;
-
-      while ((line = br.readLine()) != null) {
-        if (line.contains("{{username}}") || line.contains("{{password}}")) {
-          line = replaceUserDbProxyCredentials(line);
-        }
-
-        sb.append(line);
-        if (line.contains(";")) {
-          String query = sb.toString();
-          stmt.execute(query);
-          sb = new StringBuilder(); // reinitialize, otherwise will keep appending
-        }
+  //@Override
+  public String handleRequest(CloudFormationCustomResourceEvent customResourceEvent , Context context) {
+    boolean runScript = false;
+    String requestType = customResourceEvent.getRequestType();
+    JSONObject retJson = new JSONObject();
+    retJson.append("RequestType", requestType);
+    
+    if (null != requestType) {
+      if ("Update".equals(requestType) || "Create".equals(requestType)) {
+        // Updates and Creates will execute the sql script
+        runScript = true;
       }
-
-      return "Db Init script executed successfully";
-
-      // Rethrow the checked exception as a runtime exception as
+      if ("Update".equals(requestType) || "Delete".equals(requestType)) {
+        // Updates and Deletes need to return the same Physical Id they had
+        retJson.append("PhysicalResourceId", customResourceEvent.getPhysicalResourceId());
```

Comment:
By default, as per doc, this will be handle automatically : 

> For Update, the current physical ID will be used.

> For Delete, it will always return the current physical resource ID, and if the user returns a different one, an error will occur.

source: https://docs.aws.amazon.com/cdk/api/latest/docs/custom-resources-readme.html#handling-lifecycle-events-onevent 

I'm surprised that this  part of the change is needed